tests/perfalert/test_analyze.py

```diff
@@ -35,17 +36,18 @@ def test_calc_t(self):
 
 class TestAnalyzer(unittest.TestCase):
 
-    def test_analyze_t(self):
-        a = Analyzer()
+    def test_detect_changes(self):
+        a = []
 
         times = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
         values = [0, 0, 0, 0, 0, 0, 0, 0, 1, 1,  1,  1,  1,  1,  1,  1]
         for (t, v) in zip(times, values):
-            a.add_data(t, float(v))
+            a.append(Datum(t, float(v)))
+        a = sorted(a)
```

Comment:
It would be better to sort the array inside the analyze() function, that way callers don't have to worry about sorting it themselves.